keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -963,49 +994,18 @@ of this subsystem.  To accomplish these things there are two levels of
 defaulting: one concerning behavior, and one concerning explicit API
 objects.
 
-The effective configuration is the union of (a) the actual API objects
-that exist and (b) implicitly generated backstop objects.  The latter
-are not actual API objects, and might not ever exist as identifiable
-objects in the implementation, but are figments of our imagination
-used to describe the behavior of this subsystem.  These backstop
-objects are implicitly present and affecting behavior when needed.
-There are two implicitly generated RequestPriority backstop objects.
-One is equivalent to the `system-top` object exhibited above, and it
-exists while there is no actual RequestPriority object with `exempt ==
-true`.  The other is equivalent to the `workload-low` object exhibited
-above, and exists while there is no RequestPriority object with
-non-exempt priority.  There are also two implicitly generated
-FlowSchema backup objects.  Whenever a request whose groups include
-`system:masters` is not matched by any actual FlowSchema object, a
-backstop equivalent to the `system-top` object exhibited above is
-considered to exist.  Whenever a request whose groups do not include
-`system:masters` is not matched by any actual FlowSchema object, the
-following backstop object is considered to exist.
-
-```yaml
-kind: FlowSchema
-meta:
-  name: non-top-backstop
-spec:
-  matchingPriority: (doesn’t really matter)
-  requestPriority:
-    name: (name of an effectively existing RequestPriority, whether
-           that is real or backstop, with catchAll==true)
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: [ ] # match everything
-```
-
-The other part of the defaulting story concerns making actual API
-objects exist, and it goes as follows.  Whenever there is no actual
-RequestPriority object with `exempt == true`, the RequestPriority
-objects exhibited above are created --- except those with a name
-already in use by an existing RequestPriority object.  Whenever there
-is no actual FlowSchema object that refers to an exempt
-RequestPriority object, the schema objects shown above as examples are
-generated --- except those with a name already in use.
+The effective configuration is the union of (a) preserving configuration 
+objects and (b) all the exhibited default configurations above. The 
+system uses the preserving configuration as its initial state on 
+launching when not all the default configuration present in the system 
```

Comment:
Here I think "the system" is intended to refer to persistently stored API objects (i.e., today, in etcd).  If that is right then there is a gap here, we also need to say what happens between the time an apiserver starts handling requests and when it has had a chance to read the persistently stored API objects.

Reply:
updated